modules/api/pkg/ee/clusterbackup/storage-location/provider.go

```diff
@@ -230,6 +235,59 @@ func (p *BackupStorageProvider) Patch(ctx context.Context, userInfo *provider.Us
 	return updated, nil
 }
 
+func (p *BackupStorageProvider) ListBucketObjects(ctx context.Context, userInfo *provider.UserInfo, name string, labelSet map[string]string) (apiv2.BackupStorageLocationBucketObjectList, error) {
+	if userInfo == nil {
+		return nil, errors.New("a user is missing but required")
+	}
+
+	client := p.privilegedClient
+	if !userInfo.IsAdmin {
+		var err error
+		client, err = p.getImpersonatedClient(userInfo)
+		if err != nil {
+			return nil, err
+		}
+	}
+
+	cbsl := &kubermaticv1.ClusterBackupStorageLocation{}
+	if err := client.Get(ctx, types.NamespacedName{Name: name, Namespace: resources.KubermaticNamespace}, cbsl); err != nil {
+		return nil, common.KubernetesErrorToHTTPError(err)
+	}
+
+	cbslCredentials, err := p.getExistingCredentials(ctx, cbsl.Spec.Credential.Name)
+
+	if err != nil {
+		return nil, common.KubernetesErrorToHTTPError(err)
+	}
+
+	bucket := cbsl.Spec.ObjectStorage.Bucket
+
+	credsProvider := awsCredentials.StaticCredentialsProvider{
+		Value: aws.Credentials{
+			AccessKeyID:     cbslCredentials.AccessKeyID,
+			SecretAccessKey: cbslCredentials.SecretAccessKey,
+		},
+	}
+
+	cfg := aws.Config{
+		Region:       cbsl.Spec.Config["region"],
+		Credentials:  credsProvider,
+		BaseEndpoint: aws.String(cbsl.Spec.Config["s3Url"]),
+	}
```

Comment:
Use constants for `region` and `s3Url`.

- For region https://github.com/kubermatic/dashboard/blob/main/modules/api/pkg/ee/clusterbackup/backupstoragelocation/handler.go#L65
- For s3Url just create one

Reply:
Fixed.